### PR TITLE
Fix Custom alarm rules are overwritten by 'resource/alarm-settings.yml'(#11311)

### PR DESCRIPTION
**Pull request by ZhengBing520**
…l'(#11311)

### Fix <[bug description or the bug issue number or bug issue link](https://github.com/apache/skywalking/issues/11311)>
- [ ] Add a unit test to verify that the fix works.
- [ ] Explain briefly why the bug exists and how to fix it.

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<11311>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

### CODE REVIEW

File: oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/AlarmRulesWatcher.java [hunk 1]
```diff
@@ -54,26 +57,48 @@ public class AlarmRulesWatcher extends ConfigChangeWatcher {
...
+                Rules rules = rulesReader.readRules();
+                notify(rules);
+            }
+            notifiedByDynamicConfig.set(true);
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    public void notifyBeyondDynamicConfig(Rules newRules) {
```
Comment (wu-sheng) [inline oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/AlarmRulesWatcher.java:93]: The word beyond seems a little confused. How about initConfig directly?

Comment (ZhengBing520) [reply oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/AlarmRulesWatcher.java:93]: > The word beyond seems a little confused. How about initConfig directly?

Should I use the word 'notifyInitConfig'

Comment (wu-sheng) [reply oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/AlarmRulesWatcher.java:93]: I would say no need to add notify.

Comment (ZhengBing520) [reply oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/AlarmRulesWatcher.java]: > I would say no need to add notify.

Okay, it has been changed.

**Review by wankai123 — APPROVED** [review]

**Pull request merged**